src/auth/AuthServiceHandler.cc

```diff
@@ -26,6 +27,8 @@ AuthServiceHandler *get_auth_service_handler(int type, CephContext *cct, KeyServ
     return new CephxServiceHandler(cct, ks);
   case CEPH_AUTH_NONE:
     return new AuthNoneServiceHandler(cct);
+  case CEPH_AUTH_UNKNOWN:
+    return new AuthUnknownServiceHandler(cct);
```

Comment:
shouldn't the logic already exists? pls see https://github.com/ceph/ceph/blob/master/src/mon/AuthMonitor.cc#L428-L432.